arroyo/processing/strategies/run_task_with_multiprocessing.py

```diff
@@ -299,21 +301,104 @@ class RunTaskWithMultiprocessing(
         average message size.
 
         If the value is too small, the batch is implicitly broken up. In that
-        case, the ``batch.input.overflow`` metric is emitted.
+        case, the
+        ``arroyo.strategies.run_task_with_multiprocessing.batch.input.overflow``
+        metric is emitted.
 
     :param output_block_size: Size of the shared memory buffer used to store
         results. Like with input data, the batch is implicitly broken up on
-        overflow, and ``batch.output.overflow`` metric is incremented.
-
-        While not entirely avoidable, it is best to avoid hitting overflow too
-        often on either buffer. The performance implications of running into
-        overflow on the output buffer can be particularly significant.
+        overflow, and
+        ``arroyo.strategies.run_task_with_multiprocessing.batch.output.overflow``
+        metric is incremented.
 
     :param initializer: A function to run at the beginning of each subprocess.
 
         Subprocesses are spawned without any of the state of the parent
         process, they are entirely new Python interpreters. You might want to
         re-initialize your Django application here.
+
+    Number of processes
+    ~~~~~~~~~~~~~~~~~~~
+
+    The metric
+    ``arroyo.strategies.run_task_with_multiprocessing.batches_in_progress``
+    shows you how many processes arroyo is able to effectively use at any given
+    point.
+
+    The metric ``arroyo.strategies.run_task_with_multiprocessing.processes``
+    shows how many processes arroyo was configured with.
+
+    If those two metrics don't line up, your consumer is not bottlenecked on
+    number of processes. That's a good thing, you want to have some reserve
+    capacity. But it means that increasing ``num_processes`` will not make your
+    consumer faster.
+
+    Batching
+    ~~~~~~~~
+
+    Arroyo sends messages in batches to subprocesses. The cost of a batch is
+    amortized with larger batches. ``max_batch_size`` and ``max_batch_time``
```

Comment:
i elaborated a bit on the risks of large batches, but I am curious about this comment

> The issue is that tuning this processor is really hard and making it easy just with docs and training will not actually work.

Do you feel that batching is significantly more difficult with this processor than with other batching mechanisms? Is that because of block sizes or something else?